sky/provision/provisioner.py

```diff
@@ -442,6 +442,14 @@ def _post_provision_setup(
                                               cluster_name.name_on_cloud,
                                               provider_config=provider_config)
 
+    # Update cluster info in handle so cluster instance ids are set. This
+    # allows us to expose provision logs to debug nodes that failed during post
+    # provision setup.
+    handle = global_user_state.get_handle_from_cluster_name(
+        cluster_name.display_name)
+    handle.cached_cluster_info = cluster_info
+    global_user_state.update_cluster_handle(cluster_name.display_name, handle)
```

Comment:
I think it would be cleaner to do this in cloud_vm_ray_backend `_locked_provision`, where we are already updating the handle with other cluster_info values. https://github.com/skypilot-org/skypilot/blob/4f345c81bd4baeb1ea1e7a98dc6b650f6ec6c7c5/sky/backends/cloud_vm_ray_backend.py#L3701-L3727

Reply:
Oh I see we want this before we actually do the provisioning so that we can see the logs ;)
However, I'm a little worried, because if you look at this point in cloud_vm_ray_backend ^ we already have a reference to some handle which hasn't yet been written to the database. And we write that handle to the database later. But further down the call stack inside here we are reading and writing the handle. Seems like this change will get wiped out once the handle from cloud_vm_ray_backend gets written into the db